core/src/style/textStyle.cpp

```diff
@@ -111,6 +111,25 @@ auto TextStyle::applyRule(const DrawRule& _rule, const Properties& _props) const
         }
     }
 
+    if (!_rule.get(StyleParamKey::repeat_group, p.labelOptions.repeatGroup)) {
+        // TODO: default to 'draw.key'
+        // TODO: Optimize - this is kind of heavy on allocations
+        for (auto* name : _rule.getLayerNames()) {
+            p.labelOptions.repeatGroup += name;
+            p.labelOptions.repeatGroup += "/";
+        }
+        //LOG("rg: %s", p.labelOptions.repeatGroup.c_str());
+    }
```

Comment:
Makes sense to me; as you said except the fact that we would have a slight chance of hash collision I'm good with it.